interactions/ext/hybrid_commands/context.py

```diff
@@ -0,0 +1,386 @@
+        self.ctx = ctx
+        self.ephermal = ephermal
+
+    async def __aenter__(self) -> None:
+        await self.ctx.defer(ephemeral=self.ephermal)
+
+    async def __aexit__(self, *_) -> None:
+        pass
+
+
+class HybridContext(BaseContext, SendMixin):
+    prefix: str
+    "The prefix used to invoke this command."
+
+    app_permissions: Permissions
+    """The permissions available to this context"""
+
+    deferred: bool
+    """Whether the context has been deferred."""
+    responded: bool
+    """Whether the context has been responded to."""
+    ephemeral: bool
+    """Whether the context response is ephemeral."""
+
+    _command_name: str
+    """The command name."""
+    _message: Message | None
+
+    args: list[Any]
+    """The arguments passed to the command."""
+    kwargs: dict[str, Any]
+    """The keyword arguments passed to the command."""
+
+    __attachment_index__: int
+
+    _slash_ctx: SlashContext | None
+    _prefixed_ctx: prefixed.PrefixedContext | None
+
+    def __init__(self, client: Client):
+        super().__init__(client)
+        self.prefix = ""
+        self.app_permissions = Permissions(0)
+        self.deferred = False
+        self.responded = False
+        self.ephemeral = False
+        self._command_name = ""
+        self.args = []
+        self.kwargs = {}
+        self._message = None
+        self.__attachment_index__ = 0
+        self._slash_ctx = None
+        self._prefixed_ctx = None
+
+    @classmethod
+    def from_dict(cls, client: Client, payload: dict) -> None:
+        # this doesn't mean anything, so just implement it to make abc happy
+        raise NotImplementedError
+
+    @classmethod
+    def from_slash_context(cls, ctx: SlashContext) -> Self:
+        self = cls(ctx.client)
+        self.guild_id = ctx.guild_id
+        self.channel_id = ctx.channel_id
+        self.author_id = ctx.author_id
+        self.message_id = ctx.message_id
+        self.prefix = "/"
+        self.app_permissions = ctx.app_permissions
+        self.deferred = ctx.deferred
+        self.responded = ctx.responded
+        self.ephemeral = ctx.ephemeral
+        self._command_name = ctx._command_name
+        self.args = ctx.args
+        self.kwargs = ctx.kwargs
+        self._slash_ctx = ctx
+        return self
+
+    @classmethod
+    def from_prefixed_context(cls, ctx: prefixed.PrefixedContext) -> Self:
+        # this is a "best guess" on what the permissions are
+        # this may or may not be totally accurate
+        if hasattr(ctx.channel, "permissions_for"):
+            app_permissions = ctx.channel.permissions_for(ctx.guild.me)  # type: ignore
+        elif ctx.channel.type in {10, 11, 12}:  # it's a thread
+            app_permissions = ctx.channel.parent_channel.permissions_for(ctx.guild.me)  # type: ignore
+        else:
+            # this is what happens with interaction contexts in dms
+            app_permissions = Permissions(0b1111100110001000000)
```

Comment:
what in god's fucking name

Reply:
Can we extract the bytes to a flag, or comment on what the permissions are exactly?

Reply:
I suppose, though I think I may just make it `None` for parity with slash commands.